### PR TITLE
Gaj/feature/error loading handling

### LINKED ISSUE #14: Apply API
# Story 
As a dev, I want to use an API so that I can load data from a server for the user to view. 

## Acceptance Criteria 
When the dummy data is deleted and the API fetch is applied, the movie data appears. 

### Concerns 
- [x] Is the data in the format we expected it to be in? 
- [x] Are we passing down the data into the Film component?

### LINKED ISSUE #15: Loading Message
# Story 
As a user, I want to go to the main page and see a loading message.

## Acceptance Criteria
When we slow down web page, we as devs are able to see that the loading message appears before the API data populates the components state. 

### Concerns 
- [x] What type of UX do we want? 
- [x] Is the conditional rendering accurate? 
- [x] is what we want to be displayed being displayed?

### LINKED ISSUE #16: Error Message
# Story 
As a user, I want see an error message so that I can know when there's been a problem. 

## Acceptance Criteria 
When we the devs break the API and cause an error on purpose, the error message appears. 

### Concerns 
- [x] does the loading message disappear? 
- [x] is there no unwanted text on the page? 
- [x] Is the error messaged?

### PR DESCRIPTION
**Pull request by gaj23**
## *Purpose:*
*What does this merge do? Check all that apply.*
- [x] Adds a new feature
- [x] Improves an existing feature
- [ ] Fixes a bug
- [ ] Cleans up (logs, tests, etc.)

## *Description: What does this PR do?*
*Briefly, but clearly describe what you did, and explain any new code.*

This completes [Iteration 3](https://frontend.turing.io/projects/module-3/rancid-tomatillos-v3.html) as outlined by the Rancid Tomatillos project by Turing School of Software and Design. 

We've applied open source APIs to fetch an array of data as well as a specific object based on its ID. We've included 400 and 500 error handling messages for UX as well as loading messages. 

Additionally, we've taken the time to refactor our [ghIssues](https://github.com/gaj23/rancid/issues/) to account for acceptance criteria.

## *Instructions for reviewer:*
*Where should the reviewer start?*

The bulk of our work occurred in src/components/App and src/components/Film. 

*How should this be manually tested?*

Please break the API code by deleting or adding to the fetch url.

*Any background context you want to provide?*

No.

*What are the relevant tickets?*

Issues Closed #16
Issues Closed #15
Issues Closed #14

## *Issues:*
*Are there any concerns, issues, or bugs in this branch? If so describe them.*

As of this moment, this is best viewed in mobile setting. No mobile queries have been applied to account for desktop viewing or other screen viewing.

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have used tests to prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my change

### CODE REVIEW

**Comment by gaj23** [comment]
@sertmer & @letakeane, for our weekend deliverables! This is a **mobile first** app and best viewed in that setting.

File: src/components/App/App.js [hunk 1]
```diff
@@ -2,34 +2,52 @@ import React, { Component } from 'react';
...
   }
 
   goHome = () => {
-    this.setState({ currentMovie: {}, isHome: true})
+    this.setState({ currentMovie: '', isHome: true})
   }
 
-  selectMovie = () => {
-    this.setState({ currentMovie: {}, isHome: false });
+  selectMovie = (event) => {
```
Comment (richardltyler) [inline src/components/App/App.js:38]: Maybe it would be better to target the card by it's classname rather than the tag name in the event target

**Review by richardltyler — APPROVED** [review]
This looks awesome. Thanks for teaching me how to use conditional rendering for error-handling!

File: src/components/App/App.css [hunk 2]
```diff
@@ -1,11 +1,8 @@
 .App {
-  /* background-color: #FFFFFF; */
-  /* min-height: 100vh; */
-  /* display: flex;
+  min-height: 100vh;
```
Comment (letakeane) [inline src/components/App/App.css:2]: Don't forget to order your properties in some consistent way within the rule :)

File: src/components/App/App.js [hunk 3]
```diff
@@ -2,34 +2,52 @@ import React, { Component } from 'react';
...
 import './App.css';
 
 class App extends Component {
   constructor() {
     super();
     this.state = {
-      movies: movieData.movies,
-      currentMovie: {},
-      isHome: true
+      movies: [],
```
Comment (letakeane) [inline src/components/App/App.js:11]: Great initial values here!

File: src/components/App/App.js [hunk 4]
```diff
@@ -2,34 +2,52 @@ import React, { Component } from 'react';
...
+      currentMovie: '',
+      isHome: true,
+      isLoading: true,
+      error: false
+    }
+  }
+
+  componentDidMount() {
+    fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies/')
```
Comment (letakeane) [inline src/components/App/App.js:20]: You'll probably want to pull this (and the other network request functions) out into its own apiCalls.js file

File: src/components/App/App.js [hunk 5]
```diff
@@ -2,34 +2,52 @@ import React, { Component } from 'react';
...
+      .then(response => response.json())
+      .then(films => this.checkForError(films))
+      .catch(error => this.setState({isLoading:false, error: true}))
+  }
+
+  checkForError = films => {
```
Comment (letakeane) [inline src/components/App/App.js:26]: Love that you've put in error handling

File: src/components/App/App.js [hunk 6]
```diff
@@ -2,34 +2,52 @@ import React, { Component } from 'react';
...
+  selectMovie = (event) => {
+    this.setState({ currentMovie: event.target.closest('article').id, isHome: false });
   }
 
   render() {
     return (
       <>
         <Header goHome={this.goHome}/>
         <div className="App">
+          {this.state.isLoading && <h2>Please wait...</h2>}
```
Comment (letakeane) [inline src/components/App/App.js:47]: Nice conditional rendering

File: src/components/Film/Film.css [hunk 7]
```diff
@@ -4,12 +4,12 @@
   margin: 1rem;
   font-family: Rubik, sans-serif;
   display: grid;
```
Comment (letakeane) [inline src/components/Film/Film.css:6]: HECK YES, GRID!

File: src/components/Film/Film.js [hunk 8]
```diff
@@ -5,88 +5,111 @@ class Film extends Component {
...
-      backdrop_path: "https://image.tmdb.org/t/p/original//oazPqs1z78LcIOFslbKtJLGlueo.jpg",
-      release_date: "2019-12-04",
-      overview: "Some overview that is full of buzzwords to attempt to entice you to watch this movie! Explosions! Drama! True love! Robots! A cute dog!",
-      average_rating: 6,
-      genres: [{id: 18, name:"Drama"}],
-      budget:6300000,
-      revenue:1008537530,
-      runtime:139,
-      tagline: "It's a movie!"
+      currentMovie: {},
```
Comment (letakeane) [inline src/components/Film/Film.js:8]: Great job refactoring to use actual API data :)

**Review by letakeane — COMMENTED** [review]
Thanks for your work! The refactor is looking good.

**Pull request merged**